File: HW0/HW0-Montazeri_810699269/HW0.cpp
#include <iostream>
#include <cmath>
#include <windows.h>
#include <vector>
#include <stdlib.h>
#include <time.h>
// #include "matplotlibcpp.h"

using namespace std;
// namespace plt = matplotlibcpp;

int prob1()
{
    int x = 10;
    float y = 4.12;

    float Addition = x + y;
    float Subtraction = x - y;
    float Multiplication = x * y;
    float Division = x / y;
    float Exponentiation = exp(x);

    cout << "Addition=" << Addition;
    cout << "\nSubtraction=" << Subtraction;
    cout << "\nMultiplication=" << Multiplication;
    cout << "\nDivision=" << Division;
    cout << "\nExponentiation=" << Exponentiation;
    cout << endl;
    return 0;
}

int prob2()
{
    int age;
    cout << "Please enter your age:\t";
    cin >> age;
    if (age < 18)
    {
        cout << "you're a Minor";
    }
    else if ((18 <= age) && (age <= 65))
    {
        cout << "you're an Adult";
    }
    else if (65 < age)
    {
        cout << "you're a Senior";
    }
    else
    {
        cout << "invalid input";
    }
    cout << endl;
    return 0;
}

int prob3()
{
    int n = 12;
    int fib[n];
    int first = 0, second = 1;
    for (int i = 0; i < n; i++)
    {
        fib[i] = first;
        fib[i + 1] = second;
        int New = first + second;
        first = second;
        second = New;
    }

    for (int i = 0; i < n; i++)
    {
        cout << fib[i];
        if (i != n - 1)
        {
            cout << ", ";
        }
    }
    cout << endl;
    return 0;
}

int prob4(int n, float d, string word)
{
    for (int i = 0; i < n; i++)
    {
        cout << word;
        if (i != n - 1)
        {
            cout << " - ";
            Sleep(1000*d);
        }
    }
    cout << endl;
    return 0;
}

// int prob5()
// {
//     srand(time(0));
//     vector<float> t;
//     vector<float> wave_1, wave_2;
//     for (int i = 0; i < 1000; i++)
//     {
//         t.push_back(i / 100.0);
//         wave_1.push_back(sin(t[i]) + float(rand()) / float(RAND_MAX) * 0.1);
//         wave_2.push_back(2 * sin(1.5 * t[i]) + float(rand()) / float(RAND_MAX) * 0.35);
//     }

//     plt::figure_size(1300, 400);
//     plt::plot(t, wave_1, "r", {{"label", "y_1 = sin(t)"}});
//     plt::plot(t, wave_2, "b", {{"label", "y_2 = 2*sin(1.5t)"}});

//     plt::title("Different Sinusoidal Waves Subject to Noise");
//     plt::xlabel("time (s)");
//     plt::ylabel("Amplitude");
//     plt::grid(true);
//     plt::legend();
//     plt::show();
//     return 0;
// }


int main()
{
    cout << "\nprob1 - Arithmetic Operations\n";
    prob1();
    cout << "\nprob2 - Conditionals\n";
    prob2();
    cout << "\nprob3 - Loops\n";
    prob3();
    cout << "\nprob4 - Functions\n";
    prob4(5, 1, "Test");
    cout << "\nprob5 - Visualization";
    // prob5();
    return 0;
}

File: HW0/HW0-Montazeri_810699269/HW0.py
import numpy as np
import matplotlib.pyplot as plt
import random
import time

# prob.1 - Arithmetic Operations
print("\nprob.1 - Arithmetic Operations")
x = 10
y = 4.12

Addition = x+y
Subtraction = x-y
Multiplication = x*y
Division = x/y
Exponentiation = np.exp(x)

print(f'Addition={Addition:.2f}, Subtraction={Subtraction}, Multiplication={Multiplication}, Division={Division:.2f}, Exponentiation={Exponentiation:.2f}')


# prob.2 - Conditionals
print("\nprob.2 - Conditionals")
age = int(input("Please enter your age:\t"))
if age < 18:
    print("you're a Minor")
elif 18 <= age <= 65:
    print("you're an Adult")
elif 65 < age:
    print("you're a Senior")
else:
    print("invalid input")


# prob.3 - Loops
print("\nprob.3 - Loops")
n = 12
fib = [0, 1]
for i in range(n-2):
    new = fib[-1] + fib[-2]
    fib.append(new)
print(fib)


# prob.4 - Functions
print("\nprob.4 - Functions")

def prob4(n, d, word):
    for i in range(n):
        if i != n-1:
            end = ' - '
        else:
            end = "\n"
        print(word, end=end, flush=True)
        time.sleep(d)

prob4(5, 1, "Test")


# prob.5 - Visualization
print("\nprob.5 - Visualization")
x = np.linspace(0, 10, 1000)
noise_1 = []
noise_2 = []
for i in range(1000):
    noise_1.append(random.uniform(0, 0.05))
    noise_2.append(random.uniform(0, 0.2))

noise_1 = np.array(noise_1)
noise_2 = np.array(noise_2)

wave_1 = np.sin(x) + noise_1
wave_2 = 2*np.sin(1.5*x) + noise_2

plt.figure(1, (12, 5))
plt.plot(x, wave_1, label=r"$y_1 = \mathrm{sin}(t)$")
plt.plot(x, wave_2, label=r"$y_2 = 2\times\mathrm{sin}(1.5t)$")
plt.title("Two Different Sinusoidal Waves Subject to Noise")
plt.xlabel("time (s)")
plt.ylabel("magnitude")
plt.xlim([0, 10])
plt.legend()
plt.grid()
plt.show()
